src/processing/script/transform.js

```diff
@@ -21,33 +22,60 @@ function getChange (node, parent, key) {
     };
 }
 
-function transformChildNodes (node, changes, parentChanged) {
+function createState (currState, node, parent, key, hasTransformedAncestor) {
```

Comment:
Whoa, this gets really complicated. Can you elaborate (maybe in comment) how it actually works?